### PR TITLE
[cleanup] switch to not using apt for kubeadm, kubelet, and kubectl, curl kubeadm pre-init script

### PR DESCRIPTION
**Pull request by AshleyDumaine**
**What this PR does / why we need it**: This switches to installing the kubeadm packages with tar files intead of apt and moves kubeadm setup into a single script.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: scripts/pre-kubeadminit.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+apt-get update -y
+apt-get install -y containerd socat conntrack
+
+PATCH_VERSION=${1#[v]}
+VERSION=${PATCH_VERSION%.*}
+curl -L "https://github.com/kubernetes-sigs/cri-tools/releases/download/v${VERSION}.0/crictl-v${VERSION}.0-linux-amd64.tar.gz" | tar -C /usr/local/bin -xz
+cd /usr/local/bin
+curl -L --remote-name-all https://dl.k8s.io/release/$1/bin/linux/amd64/{kubeadm,kubelet}
+chmod +x {kubeadm,kubelet}
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
```
Comment (rahulait) [inline scripts/pre-kubeadminit.sh:86]: Should we pin kubectl to be of similar version as kubeadm?

Comment (AshleyDumaine) [reply scripts/pre-kubeadminit.sh:86]: we could though I'm not sure there's a kubectl release for every patch version of k8s. Maybe we could at least pin it for the minor version?

**Review by rahulait — APPROVED** [review]
LGTM

File: scripts/pre-kubeadminit.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+apt-get update -y
+apt-get install -y containerd socat conntrack
+
+PATCH_VERSION=${1#[v]}
+VERSION=${PATCH_VERSION%.*}
+curl -L "https://github.com/kubernetes-sigs/cri-tools/releases/download/v${VERSION}.0/crictl-v${VERSION}.0-linux-amd64.tar.gz" | tar -C /usr/local/bin -xz
+cd /usr/local/bin
+curl -L --remote-name-all https://dl.k8s.io/release/$1/bin/linux/amd64/{kubeadm,kubelet}
+chmod +x {kubeadm,kubelet}
+curl -LO "https://dl.k8s.io/release/v${VERSION}.0/bin/linux/amd64/kubectl"
```
Comment (rahulait) [inline scripts/pre-kubeadminit.sh:86]: Should we set the execute bit for kubectl as well here?

**Review by eljohnson92 — APPROVED** [review]

**Pull request merged**